src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,249 @@
+            String deadlineText = deadlineInfo[0].trim();
+            LocalDateTime deadlineDate = LocalDateTime.parse(deadlineInfo[1].trim(), formatter);
+            Deadline deadline = new Deadline(deadlineText, deadlineDate);
+            this.list.addTask(deadline);
+            this.ui.taskAddDisplay(deadline, this.list.getListLength());
+        } catch(IndexOutOfBoundsException e) {
+            String task = "deadline";
+            throw new DukeMissingArgumentException(task);
+        } catch(DateTimeParseException e) {
+            System.out.println("\tInvalid Date format (Required format: DD/MM/YYYY HH:MM)");
+        }
+    }
+
+    private void addEvent(String[] userInput) throws DukeMissingArgumentException{
+        try {
+            String eventInfo[] = userInput[1].split("/from|/to");
+            String eventText = eventInfo[0].trim();
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");
+            LocalDateTime eventFrom = LocalDateTime.parse(eventInfo[1].trim(), formatter);
+            LocalDateTime eventTo = LocalDateTime.parse(eventInfo[2].trim(), formatter);
+
+            Event event = new Event(eventText, eventFrom, eventTo);
+            this.list.addTask(event);
+            this.ui.taskAddDisplay(event, this.list.getListLength());
+        } catch (IndexOutOfBoundsException e) {
+            String task = "event";
+            throw new DukeMissingArgumentException(task);
+        } catch(DateTimeParseException e) {
+            System.out.println("\tInvalid Date format (Required format: DD/MM/YYYY HH:MM)");
+        }
+    }
+
+    private void deleteTask(String[] userInput) throws DukeTaskArgumentException, DukeMissingArgumentException, DukeInvalidArgumentsException{
+        try{
+            int taskIndex = Integer.parseInt(userInput[1]);
+            if(taskIndex > this.list.getListLength()) {
+                throw new DukeTaskArgumentException();
+            }
+            this.ui.taskDeleteDisplay(this.list, taskIndex);
+            this.list.deleteTask(taskIndex);
+            this.ui.displayTasks(this.list.getListLength());
+        } catch(IndexOutOfBoundsException e) {
+            String task = "delete";
+            throw new DukeMissingArgumentException(task);
+        } catch (NumberFormatException e) {
+            throw new DukeInvalidArgumentsException();
+        }
+    }
+
+    private void displayTasksWithDates(String[] userInput) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+        LocalDate date = LocalDate.parse(userInput[1], formatter);
+        int counter = 1;
+        for(int i = 0; i < this.list.getListLength(); i++) {
+            String taskType = this.list.getTask(i).getTaskType();
+            LocalDate taskDate = this.list.getTask(i).getDate().toLocalDate();
+            if(taskType.equals("D") || taskType.equals(("E"))){
+                if(date.equals(taskDate)) {
```

Comment:
Just a small suggestion that could improve code quality :)
```suggestion
            boolean isDeadlineOrEvent = taskType.equals("D") || taskType.equals("E");
            boolean isSameDate = date.equals(taskDate);
            if(isDeadlineOrEvent && isSameDate){
```